Donation/index.ts

```diff
@@ -1,41 +1,65 @@
-import {AzureFunction, Context, HttpRequest} from "@azure/functions"
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import { addDays, errorResponseFactory, Input } from "./handler";
 
+// POST http://localhost:7071/api/Donation
+const httpTrigger: AzureFunction = async (
+  context: Context,
+  req: HttpRequest
+): Promise<void> => {
+  context.log("HTTP trigger function processed a request.", req.body);
 
+  const input = Input.decode(req.body);
 
-// POST http://localhost:7071/api/Donation
-const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
-    context.log('HTTP trigger function processed a request.', req.body);
-    const properties = req.body.properties
-
-    const body = {
-        paymentOption: [
-            {
-                iuv: null,
-                amount: properties.amount,
-                description: properties.description,
-                isPartialPayment: false,
-                dueDate: addDays(new Date(), Number(process.env['ADD_DUE_DATE_DAYS'])),
-                retentionDate: addDays(new Date(), Number(process.env['ADD_RETENTION_DATE_DAYS'])),
-                fee: 0,
-                transfer: [{
-                    amount: properties.amount
-                }]
-            }
+  // eslint-disable-next-line no-underscore-dangle
+  if (input._tag === "Left") {
```

Comment:
doesn't use `if` statement try to use [this approach](https://github.com/pagopa/io-pagopa-proxy/blob/39a5118fd44b1b40cdf147c8a44831aa64446f02/src/controllers/restful/PaymentController.ts#L109)

Reply:
done